sklearn/linear_model/tests/test_ransac.py

```diff
@@ -90,12 +89,16 @@ def test_ransac_max_trials():
                                        random_state=0)
     assert_raises(ValueError, ransac_estimator.fit, X, y)
 
-    ransac_estimator = RANSACRegressor(base_estimator, min_samples=2,
-                                       residual_threshold=5, max_trials=11,
-                                       random_state=0)
-    assert getattr(ransac_estimator, 'n_trials_', None) is None
-    ransac_estimator.fit(X, y)
-    assert_equal(ransac_estimator.n_trials_, 2)
+    for i in range(50):
+        ransac_estimator = RANSACRegressor(base_estimator, min_samples=2,
+                                           random_state=i)
+        assert_true(getattr(ransac_estimator, 'n_trials_', None) is None)
+        ransac_estimator.fit(X, y)
+        # there is a 1e9 chance it will take these many trials. No good reason
+        # 1e2 isn't enough, can still happen
+        max_trials = _dynamic_max_trials(
+            len(X) - len(outliers), X.shape[0], X.shape[1] + 1, 1 - 1e9)
```

Comment:
I like shape[1]+1 since that's what ransac uses. I don't really mind though. I'll put it in comment